src/lib/migrations/routines/02-update-assistants-models.ts

```diff
@@ -0,0 +1,46 @@
+import type { Migration } from ".";
+import { getCollections } from "$lib/server/database";
+import { ObjectId, type AnyBulkWriteOperation } from "mongodb";
+import type { Assistant } from "$lib/types/Assistant";
+
+const updateAssistantsModels: Migration = {
+	_id: new ObjectId("5f9f3f3f3f3f3f3f3f3f3f3f"),
+	name: "Update deprecated assistants models",
+	up: async (client) => {
+		const models = (await import("$lib/server/models")).models;
+
+		const { assistants } = getCollections(client);
+
+		const modelIds = models.map((el) => el.id); // string[]
+		const defaultModelId = models[0].id;
+
+		let ops: AnyBulkWriteOperation<Assistant>[] = [];
+
+		// Find all assistants whose modelId is not in modelIds, and update it to use defaultModelId
+		for await (const assistant of assistants.find({ modelId: { $nin: modelIds } })) {
+			console.log({ assistant });
+			ops.push({
+				updateOne: {
```

Comment:
instead of updateOne and loop, could we just run a single operation:
```ts
collection.update({
	filter: { modelId: { $nin: modelIds }},
	update: { $set: { modelId: defaultModelId } },
})
```

and we wouldn't need the bulkwrite. Wdyt ?

Reply:
since we have index https://github.com/huggingface/chat-ui/blob/7bec0fbb9a9edd4b97e7700fe5fb5087b2b07cac/src/lib/server/database.ts#L148, this single `update` operation should be fine performance-wise. No?

Reply:
yeah that seems better, could it hang in a weird way if we have a lot of assistants? I think it should be safe but not 100% sure

Reply:
cc: @coyotte508

Reply:
i've implemented it, seems a lot cleaner

Reply:
handled in https://github.com/huggingface/chat-ui/pull/998/commits/a63c3d74e029aaada43bd52293f6baf31ea6a913